azurelinuxagent/ga/guestagent.py

```diff
@@ -127,17 +127,15 @@ def is_downloaded(self):
         return self.is_blacklisted or \
                os.path.isfile(self.get_agent_manifest_path())
 
-    def mark_failure(self, is_fatal=False, reason=''):
+    def mark_failure(self, is_fatal=False, reason='', report_func=event.warn):
         try:
             if not os.path.isdir(self.get_agent_dir()):
                 os.makedirs(self.get_agent_dir())
             self.error.mark_failure(is_fatal=is_fatal, reason=reason)
             self.error.save()
             if self.error.is_blacklisted:
                 msg = u"Agent {0} is permanently blacklisted".format(self.name)
-                logger.warn(msg)
-                add_event(op=WALAEventOperation.AgentBlacklisted, is_success=False, message=msg, log_event=False,
-                          version=self.version)
+                report_func(WALAEventOperation.AgentBlacklisted, msg)
```

Comment:
We need to use an event other than AgentBlacklisted for the new update logic. These failures should not get merged with the failures from the legacy logic in the Daemon.